source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOResourceResolvers.mm

```diff
@@ -0,0 +1,40 @@
+//
+//  ACOResourceResolvers.mm
+//  ACOResourceResolvers.h
+//
+//  Copyright © 2018 Microsoft. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+#import "ACOResourceResolvers.h"
+
+@implementation ACOResourceResolvers
+{
+    NSMutableDictionary<NSString *, NSObject<ACOIResourceResolver> *> *_resolvers;
+}
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _resolvers = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+- (void)setResourceResolver:(NSObject<ACOIResourceResolver> *)resolver scheme:(NSString *)scheme
+{
+    dispatch_sync(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
>dispatch_sync(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{ [](start = 4, length = 79)

I imagine this is here to synchronize access to the dictionary? But it won't work because dispatch_get_global_queue returns a concurrent queue, not a serial queue, so access is not serialized.

If you want to synchronize these methods, the easiest way is to wrap them in @synchronized(self).

Reply:
thanks; i just removed the syncronization logic since this is good to have, and user of the SDK owns the resolvers.

---
In reply to: [202422420](https://github.com/Microsoft/AdaptiveCards/pull/1702#discussion_r202422420) [](ancestors = 202422420)